src/shared/rpc/methods/send-transfer.ts

```diff
@@ -19,11 +25,27 @@ export const rpcSendTransferParamsSchemaLegacy = yup.object().shape({
 
 export const rpcSendTransferParamsSchema = yup.object().shape({
   account: accountSchema,
-  recipients: yup
-    .array()
-    .of(yup.object().shape({ address: yup.string().required(), amount: yup.string().required() }))
-    .required(),
-  network: yup.string().oneOf(Object.values(WalletDefaultNetworkConfigurationIds)),
+  network: yup.string().required().oneOf(Object.values(WalletDefaultNetworkConfigurationIds)),
+  recipients: yup.array().of(
+    yup.object().shape({
+      // check network is valid for address
+      address: btcAddressValidator().test(
+        'address-network-validation',
+        FormErrorMessages.IncorrectNetworkAddress,
+        (value, context) => {
+          const contextOptions = context.options as any;
+          const network = contextOptions.from[1].value.network as BitcoinNetworkModes;
+          return btcAddressNetworkValidator(network).isValidSync(value);
```

Comment:
magic number, what's 1?

Reply:
seems, there is no other ways to [get network here ](https://stackoverflow.com/questions/73724613/yup-validation-at-lower-level-based-on-a-higher-value-unsure-how-to-access)